src/pages/resultsView/enrichments/EnrichmentsUtil.tsx

```diff
@@ -400,4 +406,109 @@ export function getGroupColumns(groups: { name: string, description: string }[],
         });
     });
     return columns;
+}
+
+export function getEnrichmentBarPlotData(data: AlterationEnrichmentRow[], genes: string[]): IMultipleCategoryBarPlotData[] {
+    const usedMajorCategories: {[gene:string]:boolean} = {};
+    if (_.isEmpty(genes)) {
+        return [];
+    }
+    let categoryToCounts = _.reduce(data, (acc, datum) => {
+        _.each(datum.groupsSet, group => {
+            const groupName = group.name + (datum.qValue < 0.05 ? '*' : '')
+            if (genes.includes(datum.hugoGeneSymbol)) {
+                if (!acc[groupName]) {
+                    acc[groupName] = {};
+                }
+                acc[groupName][datum.hugoGeneSymbol] = group.alteredCount;
+                usedMajorCategories[datum.hugoGeneSymbol] = true;
+            }
+        });
+        return acc;
+    }, {} as { [minorCategory: string]: { [majorCategory: string]: number } });
+
+    const majorCategoryTotalCounts: { [id: string]: number } = {}
+    // ensure entries for all used minor categories - we need 0 entries for those major/minor combos we didnt see
```

Comment:
i don't get what minor/major categories means from the code.   i think it needs to be more domain specific naming